### PR TITLE
Upgrade node typings to Node 10.

### PR DESCRIPTION
**Pull request by iclanton**

### CODE REVIEW

File: apps/rush-lib/src/utilities/Utilities.ts [hunk 1]
```diff
@@ -410,7 +410,8 @@ export class Utilities {
     if (options.handleOutput) {
       Utilities._processResult(result);
     }
-    return result.status;
+
+    return result.status || 1; // If result.status is null, the process was terminated from a signal
```
Comment (octogonz) [inline apps/rush-lib/src/utilities/Utilities.ts]: build.is may have the same issue, but it's written without a compiler

File: apps/rush-lib/src/cli/RushCommandLineParser.ts [hunk 2]
```diff
@@ -355,7 +355,7 @@ export class RushCommandLineParser extends CommandLineParser {
     // performs nontrivial work that can throw an exception.  Either the Rush class would need
     // to handle reporting for those exceptions, or else _populateActions() should be moved
     // to a RushCommandLineParser lifecycle stage that can handle it.
-    if (process.exitCode > 0) {
+    if (process.exitCode! > 0) {
```
Comment (octogonz) [inline apps/rush-lib/src/cli/RushCommandLineParser.ts]: `process.exit(process.exitCode || 1)`?

File: apps/rush-lib/src/scripts/install-run.ts [hunk 3]
```diff
@@ -433,7 +433,7 @@ export function installAndRun(
     }
   );
 
-  return result.status;
+  return result.status || 1; // If result.status is null, the process was terminated from a signal
```
Comment (octogonz) [inline apps/rush-lib/src/scripts/install-run.ts]: When the executable cannot be found, IIRC the `status` will be undefined and a separate `error` field will have a JavaScript `Error` object with a meaningful message that we should maybe report.

File: apps/rush-lib/src/utilities/Utilities.ts (hunk 1 above)
Comment (iclanton) [reply apps/rush-lib/src/utilities/Utilities.ts]: I don't see `.status` referenced anywhere in any `.js` files except for `common/scripts/install-run.js`

Comment (iclanton) [reply apps/rush-lib/src/utilities/Utilities.ts]: The only `.js` file that contains `.status` is `common/scripts/install-run.js`.

Comment (octogonz) [inline apps/rush-lib/src/utilities/Utilities.ts]: Should we do this behavior instead?

```ts
  if (result.status) {
    return result.status;
  } else {
    throw result.error || new Error('An unknown error occurred.');
  }
```

File: common/changes/@microsoft/api-extractor-model/ianc-upgrade-node-typings_2020-01-15-10-32.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{
+  "changes": [
+    {
+      "comment": "Upgrade Node typings to Node 10",
+      "type": "minor",
```
Comment (octogonz) [inline common/changes/@microsoft/api-extractor-model/ianc-upgrade-node-typings_2020-01-15-10-32.json]: Maybe this should be `"patch"` -- the API doesn't seem to depend on NodeJS at all

File: common/changes/@microsoft/api-documenter/ianc-upgrade-node-typings_2020-01-15-10-32.json [hunk 5]
```diff
@@ -0,0 +1,11 @@
+{
+  "changes": [
+    {
+      "comment": "Upgrade Node typings to Node 10",
+      "type": "minor",
```
Comment (octogonz) [inline common/changes/@microsoft/api-documenter/ianc-upgrade-node-typings_2020-01-15-10-32.json]: Maybe this should be `"patch"` -- the API doesn't seem to depend on NodeJS at all

File: common/changes/@microsoft/api-extractor/ianc-upgrade-node-typings_2020-01-15-10-32.json [hunk 6]
```diff
@@ -0,0 +1,11 @@
+{
+  "changes": [
+    {
+      "comment": "Upgrade Node typings to Node 10",
+      "type": "minor",
```
Comment (octogonz) [inline common/changes/@microsoft/api-extractor/ianc-upgrade-node-typings_2020-01-15-10-32.json]: Maybe this should be `"patch"` -- the API doesn't seem to depend on NodeJS at all

File: common/changes/@microsoft/ts-command-line/ianc-upgrade-node-typings_2020-01-15-10-32.json [hunk 7]
```diff
@@ -0,0 +1,11 @@
+{
+  "changes": [
+    {
+      "comment": "Upgrade Node typings to Node 10",
+      "type": "minor",
```
Comment (octogonz) [inline common/changes/@microsoft/ts-command-line/ianc-upgrade-node-typings_2020-01-15-10-32.json]: Maybe this should be `"patch"` -- the API doesn't seem to depend on NodeJS at all

File: common/changes/@rushstack/eslint-plugin/ianc-upgrade-node-typings_2020-01-15-10-32.json [hunk 8]
```diff
@@ -0,0 +1,11 @@
+{
+  "changes": [
+    {
+      "comment": "Upgrade Node typings to Node 10",
+      "type": "minor",
```
Comment (octogonz) [inline common/changes/@rushstack/eslint-plugin/ianc-upgrade-node-typings_2020-01-15-10-32.json]: Maybe this should be `"patch"` -- the API doesn't seem to depend on NodeJS at all

File: apps/rush-lib/src/utilities/Utilities.ts (hunk 1 above)
Comment (iclanton) [reply apps/rush-lib/src/utilities/Utilities.ts]: I like that.

**Pull request merged**